git/githistory/ref_updater.go

```diff
@@ -139,11 +158,10 @@ func (r *refUpdater) updateOneRef(list *tasklog.ListTask, maxNameLen int, seen m
 		return nil
 	}
 
-	if err := git.UpdateRefIn(r.root, ref, to, ""); err != nil {
-		return err
-	}
+	*gitUpRefScript += fmt.Sprintf("update %s %s\n", ref.Refspec(), hex.EncodeToString(to))
```

Comment:
In my proposed version where we pass an input pipe to this function, and also use the NUL-delimited instruction format, this line might become:

```suggestion
	update := fmt.Sprintf("update %s\x00%s\x00\x00", ref.Refspec(), hex.EncodeToString(to))
	if _, err := io.WriteString(cmdInput, update); err != nil {
		return err
	}
```